### ISSUE TITLE
[Bug] Level failure condition does not account for loans

### ISSUE THREAD

**Issue by MRMIdAS** [CorsixTH/CorsixTH#2670]
### Describe the issue

In Vanilla TH, you could use a loan to offset the fail condition due to having negative money. Current version fails you regardless of loan amount.

### Steps to Reproduce

1. Go into debt

2. Take out loan

3. End year

### Expected Behaviour

Vanilla behaviour adds money you get from end year bonuses, then checks fail conditions (negative money, reputation, etc) 

Previous versions of Corsix would check fail conditions before adding money, not *huge* in the grand scheme of things, you just take a loan out on the 31st, and repay on the 1st.

Current behaviour appears to be checking your money, subtracting any loan you might have, then checking *that* against the fail condition. this is both incorrect and nightmarishly hard.

I suspect this ties in to the fixed bug that stops you winning levels with a loan.

### Have you enabled any current CorsixTH Experimental Features?

- [ ] Right mouse panning
- [ ] Falling actions

### Save Game

No. pointless, easily testable.

### CorsixTH Version

0.68

### Operating System

Win 10

### Theme Hospital Version

GOG.com

**Comment by ARGAMX** [comment]
> ### Describe the issue
> In Vanilla TH, you could use a loan to offset the fail condition due to having negative money. Current version fails you regardless of loan amount.

Nice discovery 👍

Can you please attach any suitable save game just to save time creating the necessary situation? Thanks.

**Comment by MRMIdAS** [comment]
Save file as requested

[Loan Test.sav.gz](https://github.com/user-attachments/files/17382335/Loan.Test.sav.gz)
or zip
[Loan.Test.sav.zip](https://github.com/user-attachments/files/17390420/Loan.Test.sav.zip)

**Comment by lewri** [comment]
> Save file as requested
> 
> [Loan Test.sav.gz](https://github.com/user-attachments/files/17382335/Loan.Test.sav.gz)

Are you able to put this into a `.zip` please so its more accessible?

**Comment by lewri** [comment]
However, I'd say this would be logical to go hand in hand with #2394 from a balance perspective. As as I know that PR didn't change anything relating to lose state, only win state.

**Comment by lewri** [comment]
Culprit is here
https://github.com/CorsixTH/CorsixTH/blob/654b888ad2997b079e684df9e0e74511a0e6c564/CorsixTH/Lua/endconditions.lua#L35-L37

@tobylane to resolve this would we just remove the custom criteria on balance? I can't remember why its there. Would EndConditions also need a reload or something.
Actually, this is a regression. 0.67 did not lose the level if you had loans to offset your balance.

**Comment by MRMIdAS** [comment]
> > Save file as requested
> > [Loan Test.sav.gz](https://github.com/user-attachments/files/17382335/Loan.Test.sav.gz)
> 
> Are you able to put this into a `.zip` please so its more accessible?

We can zip now? last time I uploaded anything here it had to be in .Tar or .Gz.

**Comment by TheCycoONE** [comment]
Related issue https://github.com/CorsixTH/CorsixTH/issues/2027

**Comment by TheCycoONE** [comment]
@lewri you should be able to use 7zip to open that.

**Comment by ARGAMX** [comment]
[Loan.Test.sav.zip](https://github.com/user-attachments/files/17390420/Loan.Test.sav.zip)

**Comment by tobylane** [comment]
The correct behaviour is to always use the balance as seen for the graph and lose checks, never with the loan removed? Taking out the balance single line that Lewri highlighted fixes that, no reload necessary. The check to prevent a win with loans is then the only time balance minus loan is calculated, 

https://github.com/CorsixTH/CorsixTH/blob/654b888ad2997b079e684df9e0e74511a0e6c564/CorsixTH/Lua/endconditions.lua#L110

The win/lose check is made in World:inEndYear, after Hospital:onEndYear, 
https://github.com/CorsixTH/CorsixTH/blob/654b888ad2997b079e684df9e0e74511a0e6c564/CorsixTH/Lua/world.lua#L1355-L1358

But also when the annual report is closed, just after the award money is given in updateAwards. Should this be the only end-of-year one, remove the previous one?
https://github.com/CorsixTH/CorsixTH/blob/b29f486101ec8d387d6f93ba8386d134dc0b9af1/CorsixTH/Lua/dialogs/fullscreen/annual_report.lua#L457-L459

**Comment by MRMIdAS** [comment]
> The correct behaviour is to always use the balance as seen for the graph and lose checks, never with the loan removed?

Yes. As it was, Corsix never checked the win/lose criteria as it was in the original game, (add end of year bonuses/punishments, then check fail conditions) instead checking fail conditions first,  this *may* have originally been because earlier versions would be quick to give reputation punishments for not curing enough people (despite curing everyone the game sent to your hospital) and for not achieving a valuable enough hospital (because screw financial responsibility) and so on, so doing it this way round gives the player a chance to rebuild lost reputation.

As it stands, the order Corsix checks ATM is fine, it's not original behaviour, but I doubt many people will care either way. As long as your money as visually presented is used, it's fine.

**Comment by TheCycoONE** [comment]
We need to come to a consensus on whether this issue is resolved or we have to move the balance check to after the awards.

**Comment by MRMIdAS** [comment]
Personally I don't mind. Have it as an option maybe? Like "Build while Paused" for example?

If I had to pick, I'd choose Original behaviour, but I'm fine with whatever is chosen.

**Comment by ARGAMX** [comment]
this issue 
0.68.0 - reproduces well
0.69.1 - does not reproduces

seems #2704 resolved this issue.

@tobylane I think we should to close this issue.

And if we need to do something more here, then we should probably create a separate issue and describe the problem there and how it can be reproduced.

Seems at the moment there is no understanding why this task is open and how the problem can be reproduced. or maybe I missed something.

**Comment by ARGAMX** [comment]
It is true that in original TH you could use a loan to offset the fail condition due to having negative money.
But original game looks like somewhat inconsistent in this regard.

Lets сheck in original TH:

I have balance -7330 and regarding to failure condition I must fail.
My money bar full red which means I must lose level.

Then I borrowed $10,000:

<img width="213" height="163" alt="Image" src="https://github.com/user-attachments/assets/9d54d519-94e1-469e-8631-4053907504c7" />

The hint text hasn't changed, and I should still lose.
But the indicator has shrunk and now doesn't reach the lose threshold.

Then I borrowed more $10,000:

<img width="214" height="163" alt="Image" src="https://github.com/user-attachments/assets/972deb6a-67fd-4232-9ef4-eb1b12599ba3" />

The hint text hasn't changed.
The indicator has shrunk even more.

Later the year is over and I haven't lost.
Why then does the tooltip text look like I should?

It seems to me that Bullfrog themselves have overdone it here.

Looks like a bug.
And perhaps the behavior in version 0.68.0 was more correct, albeit harsher.

Perhaps we could provide ability to adjust this depending on the difficulty settings chosen or something else.

But since the behavior stated in the issue is not reproducible anymore, I am closing this issue.
